lib/rails_state_machine/state_machine.rb

```diff
@@ -209,8 +209,8 @@ def revert_state
         end
 
         def prepare_state_event_change(attributes)
-          if saved_changes?
-            # After calling `save`, ActiveRecord will flag the changes that it just stored as saved.
+          if ActiveRecord::VERSION::STRING <= '5.2' && saved_changes?
```

Comment:
Even if this will probably not break in the future, `Gem::Version.new(ActiveRecord::VERSION::STRING) <= Gem::Version.new('5.2')` should be the safer way to compare versions.

Reply:
It's also slower, and since there'll never be a ActiveRecord 5.10, I'd personally prefer it the way it is.